File: Objective-C/InMobiCMP Demo - iOS/InMobiCMP Demo - iOS/ViewController.m
//
//  ViewController.m
//  InMobiCMP Demo - iOS
//
//  Copyright © 2024 InMobi. All rights reserved.
//

#import "ViewController.h"
#import "AppDelegate.h"

@interface ViewController ()
@property (weak, nonatomic) IBOutlet UIButton *forceDisplayPopup;
@property (weak, nonatomic) IBOutlet UIButton *showCCPA;
@property (weak, nonatomic) IBOutlet UIButton *showGoogleConsent;
@property (weak, nonatomic) IBOutlet UIButton *showUSRegulations;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    AppDelegate *appDelegate = (AppDelegate *)[[UIApplication sharedApplication] delegate];
    
    if(appDelegate != NULL) {
        [_forceDisplayPopup addTarget:appDelegate action:@selector(forceDisplay:) forControlEvents:UIControlEventTouchUpInside];
        [_showCCPA addTarget:appDelegate action:@selector(startCCPA:) forControlEvents:UIControlEventTouchUpInside];
        [_showGoogleConsent addTarget:appDelegate action:@selector(showGoogleConsent:) forControlEvents:UIControlEventTouchUpInside];
        [_showUSRegulations addTarget:appDelegate action:@selector(showUSRegulations:) forControlEvents:UIControlEventTouchUpInside];
    }

}


@end

File: Swift/InMobiCMP Demo - iOS/InMobiCMP Demo - iOS/ViewController.swift
//
//  ViewController.swift
//  InMobiCMP Demo - iOS
//
//  Copyright © 2024 InMobi. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    @IBOutlet weak var forceDisplayPopup: UIButton! {
        didSet {
            if let appDelegate = UIApplication.shared.delegate as? AppDelegate {
                forceDisplayPopup.addTarget(appDelegate, action: #selector(appDelegate.forceDisplayPopup), for: .touchUpInside)
            }
        }
    }
    
    @IBOutlet weak var showCCPA: UIButton! {
        didSet {
            if let appDelegate = UIApplication.shared.delegate as? AppDelegate {
                showCCPA.addTarget(appDelegate, action: #selector(appDelegate.startCCPA), for: .touchUpInside)
            }
        }
    }
    
    @IBOutlet weak var showGoogleConsent: UIButton! {
        didSet {
            if let appDelegate = UIApplication.shared.delegate as? AppDelegate {
                showGoogleConsent.addTarget(appDelegate, action: #selector(appDelegate.showGoogleConsent), for: .touchUpInside)
            }
        }
    }
    
    @IBOutlet weak var showUSRegulations: UIButton! {
        didSet {
            if let appDelegate = UIApplication.shared.delegate as? AppDelegate {
                showUSRegulations.addTarget(appDelegate, action: #selector(appDelegate.showUSRegulations), for: .touchUpInside)
            }
        }
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
    }


}

